tests/system/action/lock.py

```diff
@@ -0,0 +1,53 @@
+import threading
+from contextlib import contextmanager
+from typing import Iterator
+
+from openslides_backend.services.datastore.adapter import DatastoreAdapter
+from openslides_backend.shared.interfaces.write_request_element import (
+    WriteRequestElement,
+)
+
+pytest_thread_local = threading.local()
+
+
+@contextmanager
+def monkeypatch_datastore_adapter_write() -> Iterator[None]:
+    """
+    Use a patched DatastoreAdapter.write in this context,
+    which wraps the write with an optional lock from pytest_thread_local.testlock
+    """
+    DatastoreAdapter.write_original = DatastoreAdapter.write  # type: ignore
+    DatastoreAdapter.write = write  # type: ignore
+    try:
+        yield
+    finally:
+        DatastoreAdapter.write = DatastoreAdapter.write_original  # type: ignore
+        delattr(DatastoreAdapter, "write_original")
+
+
+def write(self, write_request_element: WriteRequestElement) -> None:  # type: ignore
+    """
+    Wraps the write of the datastore.adapter and stops a thread, if the testlock attribute is set thread-local.
+    See example in test_create_sequence_numbers_race_condition
+    """
+    threadname = (
+        pytest_thread_local.name
+        if hasattr(pytest_thread_local, "name")
+        else "No threadname"
+    )
+    if hasattr(pytest_thread_local, "testlock") and pytest_thread_local.testlock:
+        print(f"{threadname}: Acquiring")
```

Comment:
These prints are nice for debugging. But I think they should be removed, if this should be merged.

Reply:
Done